project/duties.py

```diff
@@ -20,6 +20,17 @@
 PTY = not WINDOWS and not CI
 
 
+# Detect which CI were using so we can adjust things as needed for mkdocs from site to public for Gitlab Pages
+# https://docs.gitlab.com/ee/ci/variables/predefined_variables.html
+# Available for all jobs executed in CI/CD. true when available.
+GITLAB_CI = os.environ.get("GITLAB_CI")
+
+# https://docs.github.com/en/actions/reference/environment-variables Always set to true when GitHub Actions is
+# running the workflow. You can use this variable to differentiate when tests are being run locally or by GitHub
+# Actions.
+GITHUB_CI = os.environ.get("GITHUB_ACTIONS")
```

Comment:
This is nice, but I would prefer to have this logic *templated* instead of executed at runtime. I think few users use both CIs so it's not useful to have both variables in duties.py. Also, I think both GitHub and GitLab's CI define the `CI` environment variable, so these two variables are redundant with the `CI` one (see line 18).